mithril-aggregator/src/database/provider/signer_registration.rs

```diff
@@ -0,0 +1,907 @@
+            signer_id,
+            epoch_setting_id: Epoch(epoch_setting_id_int.try_into().map_err(|e| {
+                HydrationError::InvalidData(format!(
+                    "Could not cast i64 ({epoch_setting_id_int}) to u64. Error: '{e}'"
+                ))
+            })?),
+            verification_key,
+            verification_key_signature,
+            operational_certificate,
+            kes_period: match kes_period_int {
+                Some(kes_period_int) => Some(kes_period_int.try_into().map_err(|e| {
+                    HydrationError::InvalidData(format!(
+                        "Could not cast i64 ({kes_period_int}) to u64. Error: '{e}'"
+                    ))
+                })?),
+                None => None,
+            },
+            stake: match stake_int {
+                Some(stake_int) => Some(stake_int.try_into().map_err(|e| {
+                    HydrationError::InvalidData(format!(
+                        "Could not cast i64 ({stake_int}) to u64. Error: '{e}'"
+                    ))
+                })?),
+                None => None,
+            },
+            created_at,
+        };
+
+        Ok(signer_registration_record)
+    }
+
+    fn get_projection() -> Projection {
+        let mut projection = Projection::default();
+        projection.add_field("signer_id", "{:signer_registration:}.signer_id", "text");
+        projection.add_field(
+            "epoch_setting_id",
+            "{:signer_registration:}.epoch_setting_id",
+            "integer",
+        );
+        projection.add_field(
+            "verification_key",
+            "{:signer_registration:}.verification_key",
+            "text",
+        );
+        projection.add_field(
+            "verification_key_signature",
+            "{:signer_registration:}.verification_key_signature",
+            "text",
+        );
+        projection.add_field(
+            "operational_certificate",
+            "{:signer_registration:}.operational_certificate",
+            "text",
+        );
+        projection.add_field(
+            "kes_period",
+            "{:signer_registration:}.kes_period",
+            "integer",
+        );
+        projection.add_field("stake", "{:signer_registration:}.stake", "integer");
+        projection.add_field("created_at", "{:signer_registration:}.created_at", "text");
+
+        projection
+    }
+}
+
+/// Simple [SignerRegistrationRecord] provider.
+pub struct SignerRegistrationRecordProvider<'client> {
+    client: &'client Connection,
+}
+
+impl<'client> SignerRegistrationRecordProvider<'client> {
+    /// Create a new provider
+    pub fn new(client: &'client Connection) -> Self {
+        Self { client }
+    }
+
+    fn condition_by_signer_id_and_epoch(
+        &self,
+        signer_id: String,
+        epoch: &Epoch,
+    ) -> Result<WhereCondition, StdError> {
+        let epoch = i64::try_from(epoch.0).unwrap();
+
+        Ok(WhereCondition::new(
```

Comment:
Use the `?*` syntax to ensure the index is correct and not already taken by another query parameters in case of condition composition.

Reply:
Done :heavy_check_mark: